packages/editor/src/client/atoms/comment/commentUser.tsx

```diff
@@ -0,0 +1,86 @@
+import React, {useState} from 'react'
+import {useTranslation} from 'react-i18next'
+import {Col, Drawer, Form, Row} from 'rsuite'
+
+import {FullCommentFragment, FullUserFragment, ImageRefFragment} from '../../api'
+import {ImageSelectPanel} from '../../panel/imageSelectPanel'
+import {ChooseEditImage} from '../chooseEditImage'
+import {UserSearch} from '../searchAndFilter/userSearch'
+
+interface CommentUserProps {
+  comment?: FullCommentFragment
+  setComment(comment: FullCommentFragment): void
+}
+
+export function CommentUser({comment, setComment}: CommentUserProps) {
+  const {t} = useTranslation()
+  const [open, setOpen] = useState<boolean>(false)
+
+  function setUser(user: FullUserFragment) {
+    if (!comment) {
+      return
+    }
+    setComment({...comment, user})
+  }
+
+  function setGuestUser(guestUsername: string) {
+    if (!comment) {
+      return
+    }
+    setComment({...comment, guestUsername})
```

Comment:
Hey hey, because of batching this can result in a bug. Fix could be:

```suggestion
    setComment(oldComment => ({...oldComment, guestUsername}))
```